thewalrus/quantum/__init__.py

```diff
@@ -0,0 +1,196 @@
+states, see:
+
+* Kruse, R., Hamilton, C. S., Sansoni, L., Barkhofen, S., Silberhorn, C., & Jex, I.
+  "Detailed study of Gaussian boson sampling." `Physical Review A 100, 032326 (2019)
+  <https://journals.aps.org/pra/abstract/10.1103/PhysRevA.100.032326>`_
+
+* Hamilton, C. S., Kruse, R., Sansoni, L., Barkhofen, S., Silberhorn, C., & Jex, I.
+  "Gaussian boson sampling." `Physical Review Letters, 119(17), 170501 (2017)
+  <https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.119.170501>`_
+
+* Quesada, N.
+  "Franck-Condon factors by counting perfect matchings of graphs with loops."
+  `Journal of Chemical Physics 150, 164113 (2019) <https://aip.scitation.org/doi/10.1063/1.5086387>`_
+
+* Quesada, N., Helt, L. G., Izaac, J., Arrazola, J. M., Shahrokhshahi, R., Myers, C. R., & Sabapathy, K. K.
+  "Simulating realistic non-Gaussian state preparation." `Physical Review A 100, 022341 (2019)
+  <https://journals.aps.org/pra/abstract/10.1103/PhysRevA.100.022341>`_
+
+
+Fock states and tensors
+-----------------------
+
+.. autosummary::
+
+    pure_state_amplitude
+    state_vector
+    density_matrix_element
+    density_matrix
+    fock_tensor
+    probabilities
+    update_probabilities_with_loss
+    update_probabilities_with_noise
+    normal_ordered_expectation
+    fidelity
+
+Details
+^^^^^^^
+
+.. autofunction::
+    pure_state_amplitude
+
+.. autofunction::
+    state_vector
+
+.. autofunction::
+    density_matrix_element
+
+.. autofunction::
+    density_matrix
+
+.. autofunction::
+    fock_tensor
+
+.. autofunction::
+    probabilities
+
+.. autofunction::
+    update_probabilities_with_loss
+
+.. autofunction::
+    update_probabilities_with_noise
+
+.. autofunction::
+    normal_ordered_expectation
+
+.. autofunction::
+    fidelity
+
+
+Utility functions
+-----------------
+
+.. autosummary::
+
+    reduced_gaussian
+    Xmat
+    Qmat
+    Covmat
+    Amat
+    Beta
+    Means
+    prefactor
+    find_scaling_adjacency_matrix
+    mean_number_of_clicks
+    mean_number_of_clicks_graph
+    find_scaling_adjacency_matrix_torontonian
+    gen_Qmat_from_graph
+    photon_number_mean
+    photon_number_mean_vector
+    photon_number_covar
+    photon_number_covmat
+    is_valid_cov
+    is_pure_cov
+    is_classical_cov
+    find_classical_subsystem
+    total_photon_num_dist_pure_state
+    gen_single_mode_dist
+    gen_multi_mode_dist
+    normal_ordered_complex_cov
+
+
+Details
+^^^^^^^
+"""
+
+from .fock_states_and_tensors import (
+    pure_state_amplitude,
+    state_vector,
+    density_matrix_element,
+    density_matrix,
+    fock_tensor,
+    probabilities,
+    update_probabilities_with_loss,
+    update_probabilities_with_noise,
+    fidelity,
+
+    find_classical_subsystem,
+    prefactor,
+    loss_mat,
+)
+
+from .adjacency_matrices import (
+    mean_number_of_clicks,
+    mean_number_of_clicks_graph,
+    variance_number_of_clicks,
+    find_scaling_adjacency_matrix,
```

Comment:
maybe the `*_clicks` functions could go to live in the `means_and_variances` submodule or have their own?